protocol/contracts/templegold/DaiGoldAuction.sol

```diff
@@ -0,0 +1,341 @@
+
+    /// @notice Auction duration
+    uint64 public constant AUCTION_DURATION = 1 weeks;
+    /// @notice Configuration for auction
+    AuctionConfig private auctionConfig;
+
+    constructor(
+        address _templeGold,
+        address _bidToken,
+        address _treasury,
+        address _rescuer,
+        address _executor,
+        address _auctionStarter
+    ) TempleElevatedAccess(_rescuer, _executor) {
+        templeGold = ITempleGold(_templeGold);
+        bidToken = IERC20(_bidToken);
+        treasury = _treasury;
+        auctionStarter = _auctionStarter;
+        emit AuctionStarterSet(_auctionStarter);
+    }
+
+    /**
+     * @notice Set auction configuration
+     * @param _config Auction configuration
+     */
+    function setAuctionConfig(AuctionConfig calldata _config) external override onlyElevatedAccess {
+        if (_config.auctionStartCooldown == 0
+                || _config.auctionMinimumDistributedGold == 0
+                || _config.auctionsTimeDiff == 0) 
+            { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        uint256 currentEpochCache = _currentEpochId;
+        if (!epochs[currentEpochCache].hasEnded()) { revert InvalidOperation(); }
+        auctionConfig = _config;
+
+        emit AuctionConfigSet(currentEpochCache, _config);
+    }
+
+    /**
+     * @notice Set address to trigger auction start. Zero address accepted
+     * @param _starter Auction starter
+     */
+    function setAuctionStarter(address _starter) external override onlyElevatedAccess {
+        /// @notice No zero address checks. Zero address is a valid input
+        auctionStarter = _starter;
+        if (!epochs[_currentEpochId].hasEnded()) { revert InvalidOperation(); }
+        emit AuctionStarterSet(_starter);
+    }
+
+    /**
+     * @notice Set token used for bidding
+     * @param _bidToken Bid token
+     */
+    function setBidToken(address _bidToken) external override onlyElevatedAccess {
+        if (_bidToken == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        if (!epochs[_currentEpochId].hasEnded()) { revert InvalidOperation(); }
+        bidToken = IERC20(_bidToken);
+        emit BidTokenSet(_bidToken);
+    }
+
+    /**
+     * @notice Start auction. Auction start can be triggered by anyone if `auctionStarter` not set
+     * @dev The Temple Gold amount for the auction is fixed and set at startAuction(). 
+     * So in `startAuction()`, there is a call to `_distributeGold()` to mint and distribute TGOLD. 
+     * Any other `_distributeGold()` calls during auction is tracked for next auction use.
+     */
+    function startAuction() external override {
+        if (auctionStarter != address(0) && msg.sender != auctionStarter) { revert CommonEventsAndErrors.InvalidAccess(); }
+        uint256 currentEpochCache = _currentEpochId;
+        EpochInfo storage prevAuctionInfo = epochs[currentEpochCache];
+        if (!prevAuctionInfo.hasEnded()) { revert CannotStartAuction(); }
```

Comment:
There is no requirement that the auction configs have been set. The auctions could be started with all the `AuctionConfig` fields set to zero, which would have the following consequences:
- auctionsTimeDiff = 0  >>> startAuction() can be called right away
- auctionStartCooldown = 0  >>> auction starts right after startAuction is called
- auctionMinimumDistributedGold = 0  >>> an auction can start even if there are no TempleGold to distribute, which makes the auction pretty useless. 

I'm aware you probably won't start an auction without configs but if `auctionStarter == address(0)` , anyone could do it (by mistake or as a greifing attack, because I don't see any economic incentives). So either include a small `require`, or don't set `auctionStarter=address(0)` until configs have been set. 

Important to note that if this does happen, it limits significantly the time period where the mistake can be fixed and new configs can be added, so it might require a new contract deployment, but I don't see user funds at risk anywhere. 

Small proof of code that the auction can actually be started without configs that can be added to `test/forge/templegold/DaiGoldAuction.t.sol`:

```solidity
contract DaiGoldAuction_POCs_Test is DaiGoldAuctionTestBase {

    function test_startAuction_canBeCalledBeforeConfigsAreSet() public {
        IDaiGoldAuction.AuctionConfig memory config = daiGoldAuction.getAuctionConfig();

        assertEq(config.auctionsTimeDiff, 0);
        assertEq(config.auctionStartCooldown, 0);
        assertEq(config.auctionMinimumDistributedGold, 0);

        vm.prank(executor);
        daiGoldAuction.startAuction();

        // we can see that the startTime is already set by reading the auction info
        IAuctionBase.EpochInfo memory info = daiGoldAuction.getEpochInfo(1);
        assertEq(info.startTime, block.timestamp);
        
        // as the cooldown is 0, bids can happen right away
        vm.startPrank(alice);
        bidToken.approve(address(daiGoldAuction), type(uint).max);
        daiGoldAuction.bid(100 ether);
    }
}
```

@frontier159 I have made a proof of code, but I'm not too keen on labeling this one **medium** because of how unlikely it is that it would happen, and the fact that there are no user funds at risk.

Reply:
I'm not sure how the `_distributeGold()` prevents this, can you elaborate?

If configs are not set, and `totalGoldAmount == 0` (which can happen if in TempleGold `mintAmount < MINIMUM_MINT` inside `_canDistribute()`, then the following condition doesn't revert:
```solditiy
if (totalGoldAmount < config.auctionMinimumDistributedGold) { revert LowGoldDistributed(totalGoldAmount); }
```

Again, the conditions required for this to happen are very unlikely.